### PR TITLE
Flavius initial review

### PR DESCRIPTION
**Pull request by FlaviusMD**
* Fixed docker bug. Now we are querying the user's node child process to get where docker is running
* Modified logic to only start creating a github repo if everything is running smoothly (all params passed in, docker found)
* Modifies package json to start project from index.ts where it was expected. 
* Refactored Build class in its own file in the lib folder.

### CODE REVIEW

File: cli.ts [hunk 1]
```diff
@@ -69,7 +69,61 @@ function askQuestion(query: string): Promise<string> {
...
+//   await project.buildAndPush({ debug })
+
+//   if (!offline) {
+//     let { text } = project.completion
+//     await writeFile("./build/completion.json", text!)
+//   }
+// })().catch((err) => {
+//   console.error(err)
+//   process.exit(1)
+// })
```
Comment (jamesmurdza) [inline cli.ts]: We can delete this, right?

File: github.ts [hunk 2]
```diff
@@ -105,7 +105,7 @@ async function addGitHubCollaborator(token: string, repoName: string, collaborat
   };
 
   const response = await fetch(`https://api.github.com/repos/${repoName}/collaborators/${collaborator}`, requestOptions);
-  if (response.status === 204) {
+  if (response.status === 204) { // TODO Does this work? Stumbled in the past upon systems that were returning simply a 200 for creation, instead of 204. Myabe better to check for general success of 2xx?
```
Comment (jamesmurdza) [inline github.ts]: Good point. Let's do

`response.status >= 200 && response.status < 300`

to make this code more correct?

File: lib/build.ts
Comment (jamesmurdza) [inline lib/build.ts:1]: I like this. Should we move other files to the lib directory? The flat structure is a bit too much.

File: lib/build.ts [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+        console.log("Completion received. (2/2)");
+
+        return this.completion
+    }
+
+    buildAndPush = async ({ debug = false, onStatusUpdate = async ({ }) => { }, } = {}) => {
+
+        // This function pushes a status update to the database.
+        const updateStatus = async ({ finished = false } = {}) => {
+        await onStatusUpdate({ // TODO understand the purpose of this.
```
Comment (jamesmurdza) [inline lib/build.ts]: In the web UI, this is used to provide incremental updates to the user.

File: lib/build.ts [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+import { newProjectPrompt, changeProjectPrompt, planChangesPrompt } from "../prompt"
+import { createGitHubRepo, addGitHubCollaborator, correctBranchName } from "../github"
+import * as scripts from "../scripts"
+import { BuildPlan } from "../buildPlan"
+
+const { exec } = require('child_process');
+
+// TODO remove these as now env gets initialised on index.ts
+import * as dotenv from "dotenv"
+dotenv.config()
```
Comment (jamesmurdza) [inline lib/build.ts]: We can delete this.

**Review by jamesmurdza — CHANGES_REQUESTED** [review]
Thank you Flavius!

**Review by gitwitdev — APPROVED** [review]
Great stuff! I love it. 🫶

**Comment by jamesmurdza** [comment]
> Great stuff! I love it. 🫶

Me too.

**Pull request merged**